db/migrations/V2.14.0__Add_unique_constraints.sql

```diff
@@ -0,0 +1,82 @@
+-- Do not allow multiple unknown streets per kunta
+DELETE from jkr_osoite.katu duplicate
+    WHERE exists (
+        select 1 from jkr_osoite.katu where
+            katu.katunimi_fi is null and duplicate.katunimi_fi is null and
+            katu.katunimi_sv is null and duplicate.katunimi_sv is null and
+            katu.kunta_koodi = duplicate.kunta_koodi
+            and katu.id < duplicate.id
+    );
+CREATE UNIQUE INDEX idx_katu_kunta
+    ON jkr_osoite.katu
+    (kunta_koodi)
+    WHERE katu.katunimi_fi is null and katu.katunimi_sv is null;
+
+-- All osoite fields are nullable. Most addresses have all fields.
+-- Is not distinct from is too slow, better delete null and non-null
+-- duplicates using separate queries.
+DELETE from jkr.osoite duplicate
```

Comment:
I'm not sure if I'm right but could we just use a singe delete statement instead of these all separate deletes before each index creation?
I mean if we check all columns and delete duplicates does that cover all cases?
Now when I wrote that I may remember seeing some comment of yours about this decision.

Here are two alternative ways to delete the duplicates:
```sql
DELETE
FROM jkr.osoite duplicate
USING jkr.osoite
WHERE
    osoite.osoitenumero is not distinct from duplicate.osoitenumero
    and osoite.katu_id is not distinct from duplicate.katu_id
    and osoite.rakennus_id is not distinct from duplicate.rakennus_id
    and osoite.posti_numero is not distinct from duplicate.posti_numero
    and osoite.id < duplicate.id
);

DELETE
FROM jkr.osoite
WHERE id IN (
  SELECT id FROM (
    SELECT id, row_number() over (partition by rakennus_id, katu_id, osoitenumero, posti_numero order by id) from jkr.osoite
  ) duplikaatit
  WHERE duplikaatit.row_number > 1
);
```

Reply:
Yeah, the `is not distinct` query is too slow, that's why I split it to separate deletes. Will check the latter one though.

Reply:
In my opinion, the latter query will delete all rows if they are duplicated. We want to iterate through the rows so that one of each will remain, that's why the `osoite.id < duplicate.id` condition?

Reply:
The row_number() returns a row number for a row inside each group separately. From each group it keeps the one with has row number 1 and delete rows with greater row number aka duplicates.